src/core/entity.js

```diff
@@ -246,12 +227,17 @@ class Entity {
     }
 
     if (this.keyboard_controlled && this.game) {
-      Object.keys(this.dir_desc).forEach((key) => {
-        if (this.dir_desc[key]) {
-          this.dir[this.dir_desc[key]] = this.game.keys[key] || false;
-        }
-      });
-      this.do_step(tick_length);
+      const current_dir = {};
+
+      for (const [key_code, dir] of Object.entries(DIR_KEYS)) {
```

Comment:
`Object.entries` is [well-supported](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/entries#Browser_compatibility) in Firefox, Chrome and Safari.

Reply:
Wow, never heard of this tbh.